### PR TITLE
Pmd7 upgrade 7.2.0

### PR DESCRIPTION
**Pull request by jborgers**

### CODE REVIEW

**Comment by stokpop** [comment]
The PR says to merge into `master`, but should be based on and merged into `pmd7` branch?

File: src/main/resources/category/java/common.xml [hunk 1]
```diff
@@ -5,35 +5,26 @@
...
           externalInfoUrl="${doc_root}/JavaCodePerformance.md#pml05">
         <description>Problem: A proxy object is created by Contexts and Dependency Injection (CDI) for explicit references, they are not de-referenced implicitly and become a memory leak. &#13;
             Solution: Destroy the reference explicitly.
         </description>
         <priority>1</priority>
         <properties>
-            <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
-            <property name="tag" value="performance" type="String" description="for-sonar"/>
-            <!--property name="version" value="2.0"/-->
+            <property name="pmd-version" value="7" type="String" description="for-compatibility"/>
```
Comment (stokpop) [inline src/main/resources/category/java/common.xml:16]: is this property just informational or is used elsewhere?

File: src/main/resources/category/java/common.xml [hunk 2]
```diff
@@ -5,35 +5,26 @@
...
-PrimaryPrefix/Name/@Image='CDI.current'][PrimarySuffix/@Image = 'destroy']
-[PrimarySuffix/Arguments/ArgumentList/Expression/PrimaryExpression/PrimaryPrefix/Name/@Image =
-ancestor::MethodDeclaration//VariableInitializer/Expression/PrimaryExpression/PrimaryPrefix/Name[@Image='CDI.current']/../../
-PrimarySuffix[@Image = 'select']/ancestor::VariableDeclarator/VariableDeclaratorId/@Name
-])]
+(ancestor::MethodDeclaration//TryStatement/FinallyClause
+    //MethodCall[@MethodName='destroy'][ArgumentList/VariableAccess[@Name=ancestor::
+        MethodDeclaration//(VariableDeclarator|AssignmentExpression)[.//MethodCall[@MethodName='current']/TypeExpression/ClassType[@SimpleName='CDI']]/(VariableId|VariableAccess)/@Name]]
+    [MethodCall[@MethodName='current']/TypeExpression/ClassType[@SimpleName='CDI']])
+]
```
Comment (stokpop) [inline src/main/resources/category/java/common.xml:27]: Seems more concise and shorter for pmd7... useful it seems: `MethodCall[@MethodName]` etc...
Good to have unit tests to migrate :-)

**Comment by stokpop** [comment]
Looking at this PR I found this: 
* https://pmd.github.io/pmd/pmd_userdocs_migrating_to_pmd7.html
* https://pmd.github.io/pmd/pmd_userdocs_migrating_to_pmd7.html#general-ast-changes-to-avoid-image
Guess you have used these pages? It makes a suggestion to move from 7.0.0-rc1 to rc2 to rc3 ... etc to do migration in steps. Is that necessary/useful?

**Comment by stokpop** [comment]
note: `src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml` seems to contain *all* xml files

should all be deleted and only contain the migrated ones? they are also in `test-todo` branch

**Review by stokpop — CHANGES_REQUESTED** [review]
See also comments and questions in the PR conversation.

File: src/main/resources/category/java/common.xml (hunk 1 above)
Comment (jborgers) [reply src/main/resources/category/java/common.xml:16]: just informational, at least for now

File: src/main/resources/category/java/common.xml (hunk 2 above)
Comment (jborgers) [reply src/main/resources/category/java/common.xml:27]: yes, indeed :-)

**Comment by jborgers** [comment]
> Looking at this PR I found this:
> 
> * https://pmd.github.io/pmd/pmd_userdocs_migrating_to_pmd7.html
> * https://pmd.github.io/pmd/pmd_userdocs_migrating_to_pmd7.html#general-ast-changes-to-avoid-image
>   Guess you have used these pages? It makes a suggestion to move from 7.0.0-rc1 to rc2 to rc3 ... etc to do migration in steps. Is that necessary/useful?
> 
> Is the remark about use of DomXPathRule valid for us? https://pmd.github.io/pmd/pmd_userdocs_migrating_to_pmd7.html#xml-and-pom Did not see that class, is that 7.3.0-SNAPSHOT specific?

DomXPathRule seems an improvement indeed! You can access comments! To try out

**Comment by jborgers** [comment]
> note: `src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml` seems to contain _all_ xml files
> 
> should all be deleted and only contain the migrated ones? they are also in `test-todo` branch

Yes, indeed, it worked without removal, but they should be removed, done.

**Pull request merged**